src/AWS.Deploy.Orchestration/DeploymentManifestEngine.cs

```diff
@@ -0,0 +1,264 @@
+using System.Collections.Generic;
+using System.IO;
+using System.Threading.Tasks;
+using AWS.Deploy.Common.DeploymentManifest;
+using AWS.Deploy.Common.IO;
+using Newtonsoft.Json;
+using System.Linq;
+using AWS.Deploy.Common;
+using System.Diagnostics;
+
+namespace AWS.Deploy.Orchestration.DeploymentManifest
+{
+    public interface IDeploymentManifestEngine
+    {
+        Task UpdateSaveCdkProject(string saveCdkDirectoryFullPath);
+        Task UpdateLastDeployedStack(string stackName);
+        Task<DeploymentManifestModel?> GetDeploymentManifest();
+        Task DeleteLastDeployedStack(string stackName);
+        Task<List<string>> GetRecipeDefinitionPaths();
+        Task CleanOrphanStacks(List<string> deployedStacks);
+    }
+
+    /// <summary>
+    /// This class contains the helper methods to update the deployment manifest file
+    /// that keeps track of the save CDK deployment projects.
+    /// </summary>
+    public class DeploymentManifestEngine : IDeploymentManifestEngine
+    {
+        private readonly IDirectoryManager _directoryManager;
+        private readonly IFileManager _fileManager;
+        private readonly OrchestratorSession _session;
+        private readonly string _targetApplicationFullPath;
+
+        private const string DEPLOYMENT_MANIFEST_FILE_NAME = "aws-deployments.json";
+
+        public DeploymentManifestEngine(IDirectoryManager directoryManager, IFileManager fileManager, OrchestratorSession session, string targetApplicationFullPath)
+        {
+            _directoryManager = directoryManager;
+            _fileManager = fileManager;
+            _session = session;
+            _targetApplicationFullPath = targetApplicationFullPath;
+        }
+
+        /// <summary>
+        /// This method updates the deployment manifest json file by adding the directory path at which the CDK deployment project is saved.
+        /// If the manifest file does not exists then a new file is generated.
+        /// <param name="saveCdkDirectoryFullPath">The absolute path to the directory at which the CDK deployment project is saved</param>
+        /// <exception cref="FailedToUpdateDeploymentManifestFileException">Thrown if an error occured while trying to update the deployment manifest file.</exception>
+        /// </summary>
+        public async Task UpdateSaveCdkProject(string saveCdkDirectoryFullPath)
+        {
+            try
+            {
+                if (!_directoryManager.Exists(saveCdkDirectoryFullPath))
+                    return;
+
+                var deploymentManifestModel = await GetDeploymentManifest();
+                var targetApplicationDirectoryPath = _directoryManager.GetDirectoryInfo(_targetApplicationFullPath).Parent.FullName;
+                var saveCdkDirectoryRelativePath = _directoryManager.GetRelativePath(targetApplicationDirectoryPath, saveCdkDirectoryFullPath);
+
+                if (deploymentManifestModel != null && deploymentManifestModel.DeploymentManifestEntries != null)
```

Comment:
If `deploymentManifestModel` is not null then you should always use that object not just if `deploymentManifestModel.DeploymentManifestEntries` is not null. Otherwise you are wiping out the rest of the data in the `deploymentManifestModel` the first time something is added to `DeploymentManifestEntries`.